api/spec/src/app/marketplace.tsp

```diff
@@ -0,0 +1,164 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/openapi3";
+
+import "../errors.tsp";
+import "../oauth.tsp";
+import "../pagination.tsp";
+import "../types.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace OpenMeter.App;
+
+/**
+ * Marketplace API.
+ */
+@route("/api/v1/marketplace")
+@tag("Apps")
+interface Marketplace {
+  /**
+   * List available apps of the app marketplace.
+   */
+  @get
+  @route("/listings")
+  @operationId("marketplaceListListings")
+  marketplaceListings(...PaginatedQuery): Paginated<Listing> | CommonErrors;
+
+  /* **************** Install App via OAuth2 **************** */
+
+  /**
+   * Install an app via OAuth.
+   * Returns a URL to start the OAuth 2.0 flow.
+   */
+  @get
+  @route("/install/oauth2")
+  @operationId("marketplaceOAuth2InstallGetURL")
+  marketplaceOAuth2InstallGetURL(@query appKey: Key): OAuth2.ClientAppStartResponse | CommonErrors;
+
+  /**
+   * Authorize OAuth2 code.
+   * Verifies the OAuth code and exchanges it for a token and refresh token
+   */
+  @get
+  @route("/install/oauth2/authorize")
+  @operationId("marketplaceOAuth2InstallAuthorize")
+  marketplaceOAuth2InstallAuthorize(
+    ...MarketplaceOAuth2InstallAuthorizeRequest,
+  ): MarketplaceOAuth2InstallAuthorizeResponse | CommonErrors;
```

Comment:
I could be wrong here, but this isn't a public API per se, is it?

Reply:
The OAuth provider calls it with the authorization token, we exchange it for refresh token and redirect from it. So it's public.